### ISSUE TITLE
add file manipulation config options

### ISSUE THREAD

**Issue by michel-laterman** [elastic/fleet-server#3515]
File upload endpoints have hard coded configuration options:
https://github.com/elastic/fleet-server/blob/main/internal/pkg/api/handleUpload.go#L33-L34

These options should be configurable through fleet-server's policy (max file size, time limit).

Note the max file size limit is currently shared between upload/download endpoints, we will want to split these into two options

**Comment by nimarezainia** [comment]
@ycombinator @cmacknz this probably needs a technical discussion, I am wondering where this UI should be exposed? If anything I believe it should be on the Fleet Server config (settings tab). and not the Fleet server policy (as that is another agent policy).

**Comment by cmacknz** [comment]
Yeah the settings tab makes more sense than the Fleet server policy, that is where similar things live.

**Comment by cmacknz** [comment]
- https://github.com/elastic/kibana/issues/183692

The reason we filed this is because we observed users with diagnostics upload timeouts, but this may actually have been because of the bug above where diagnostics requests were limited to 3 minutes.
